api/model/src/main/java/org/projectnessie/model/MergeBehavior.java

```diff
@@ -15,11 +15,36 @@
  */
 package org.projectnessie.model;
 
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import java.io.IOException;
+import java.util.Locale;
+
 public enum MergeBehavior {
   /** Keys with this merge mode will be merged, conflict detection takes place. */
   NORMAL,
   /** Keys with this merge mode will be merged unconditionally, no conflict detection. */
   FORCE,
   /** Keys with this merge mode will not be merged. */
-  DROP
+  DROP;
+
+  public static MergeBehavior parse(String mergeBehavior) {
+    try {
+      if (mergeBehavior != null) {
+        return MergeBehavior.valueOf(mergeBehavior.toUpperCase(Locale.ROOT));
+      }
+      return null;
+    } catch (IllegalArgumentException e) {
+      return NORMAL;
```

Comment:
I'd add an `UNDEFINED` case for this. It is not likely that client gets a merge behaviour value in a response that it did not request, but in case it happens, I guess we should not try to make it look like it was `NORMAL`.

Reply:
`UNDEFINED` could also make sense as a value for `BaseMergeTransplant.getDefaultKeyMergeMode()` to indicate that the client expects all affected keys to be covered by `getKeyMergeModes()` and the server should error out if it has to fallback to `getDefaultKeyMergeMode()` for some reason.

Reply:
Hm - in that case it's maybe better to not add a "fallback value" to `MergeBehavior`, because it's first sent to the server. If the server doesn't know about it, it should definitely error out.

Reply:
If we're sure the server will not use new values in merge responses - then yes.

Reply:
Yes, it replies with what the client sent.